internal/controller/numaflowcontrollerrollout_controller.go

```diff
@@ -352,38 +368,27 @@ func (r *NumaflowControllerRolloutReconciler) getResourceOperations() (kubeUtil.
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *NumaflowControllerRolloutReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	controller, err := runtimecontroller.New(ControllerNumaflowControllerRollout, mgr, runtimecontroller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
 
-	return ctrl.NewControllerManagedBy(mgr).
-		// Reconcile NumaflowControllerRollouts when there's been a Generation changed (i.e. Spec change)
-		For(&apiv1.NumaflowControllerRollout{}).WithEventFilter(predicate.GenerationChangedPredicate{}).
-		Complete(r)
-
-	/*
-	   controller, err := runtimecontroller.New(ControllerNumaflowControllerRollout, mgr, runtimecontroller.Options{Reconciler: r})
-
-	   	if err != nil {
-	   		return err
-	   	}
-
-	   // Watch NumaflowControllerRollouts
-
-	   	if err := controller.Watch(source.Kind(mgr.GetCache(), &apiv1.NumaflowControllerRollout{}), &handler.EnqueueRequestForObject{}, predicate.GenerationChangedPredicate{}); err != nil {
-	   		return err
-	   	}
-
-	   // Watch Deployments of numaflow-controller
-	   // Can add other resources as well
-	   numaflowControllerDeployments := appv1.Deployment{}
-	   numaflowControllerDeployments.Name = "numaflow-controller" // not sure if this would work or not
-	   if err := controller.Watch(source.Kind(mgr.GetCache(), &numaflowControllerDeployments),
+	// Watch NumaflowControllerRollouts
+	if err := controller.Watch(source.Kind(mgr.GetCache(), &apiv1.NumaflowControllerRollout{}), &handler.EnqueueRequestForObject{}, predicate.GenerationChangedPredicate{}); err != nil {
+		return err
+	}
 
-	   		handler.EnqueueRequestForOwner(mgr.GetScheme(), mgr.GetRESTMapper(), &apiv1.NumaflowControllerRollout{}, handler.OnlyControllerOwner()),
-	   		predicate.GenerationChangedPredicate{}); err != nil {
-	   		return err
-	   	}
+	// Watch Deployments of numaflow-controller (NOTE: could watch other resources as well)
```

Comment:
nit: this comment is not quite right? as we are watching `numaflow-controller `deployment only?

Reply:
I meant that that we are watching the numaflow-controller deployment but we could watch additional resources as needed. I'll change the comment a bit to clarify.